app/src/main/java/com/mionix/myapplication/MainActivity.kt

```diff
@@ -18,31 +17,31 @@ import android.widget.EditText
 import androidx.appcompat.app.AppCompatActivity
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
-import androidx.room.Room
 import com.mionix.myapplication.DB.DataTable
 import com.mionix.myapplication.DB.LocalDB
 import kotlinx.android.synthetic.main.activity_main.*
-import kotlinx.android.synthetic.main.dialog_custom_layout.*
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.*
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 
 class MainActivity : AppCompatActivity() {
     //load more
-    private var page = 1
     private var isLoading = true
```

Comment:
Nhét isLoading vô ViewModel thì sao?

Reply:
isloading với  is atozfilter là diễn tả trạng thái của view mà :) sao đẩy xuống viewmodel làm gì